packages/frontend/src/components/wallet-migration/modals/CleanKeysModal/AccessKeyList.jsx

```diff
@@ -0,0 +1,127 @@
+
+const AccessKeyListContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+    text-align: left;
+
+    .access-key {
+        display: flex;
+        flex: 1;
+        font-weight: 600;
+        padding: 16px 0;
+        border-bottom: 1px solid lightgray;
+
+        .public-key {
+            flex: 7;
+
+            .key-prefix {
+                font-family: monospace;
+            }
+        }
+
+        .remove-checkbox {
+            flex: 1;
+        }
+    }
+
+    .account {
+        background-color: #fafafa;
+        border-bottom: 1px solid #f0f0f0;
+        display: flex;
+        flex: 1;
+        flex-direction: row;
+        width: 100%;
+
+        .account-detail {
+            flex: 7;
+
+            .account-id {
+                font-weight: bold;
+                text-overflow: ellipsis;
+            }
+
+            .keys-to-remove {
+                color: #de2e32;
+            }
+        }
+
+        .expand-keys {
+            flex: 1;
+            padding: 16px 0 0 0;
+        }
+    }
+`;
+
+const AccessKeyList = ({ account, onClose, onNext, selectKey, selectedKeys }) => {
+    const [expanded, setExpanded] = useState(true);
+    return (
+        <AccessKeyListContainer>
+            <h3 className='title'>
+                <Translate id='walletMigration.cleanKeys.accountTitle' />
+            </h3>
+            <div
+                className={classNames(['account', expanded ? 'open' : ''])}
+                id='full-access-keys'
+                onClick={() => setExpanded(!expanded)}
+            >
+                <div className='account-detail'>
+                    <p className='account-id'>{account.accountId}</p>
+                    <p className='keys-to-remove'>
+                        {account.accessKeys.length}/{account.totalAccessKeys}
+                        <Translate id='walletMigration.cleanKeys.fullAccessKeys' />
+                    </p>
+                </div>
+                <div className='expand-keys'>
+                    <ChevronIcon color='#0072ce' />
+                </div>
+            </div>
+            <Accordion
+                trigger='full-access-keys'
+                className='breakdown'
+            >
+                {account.accessKeys.map(({ kind, publicKey }) => (
+                    <div className='access-key' key={publicKey}>
+                        <div className='public-key'>
+                            <span className='key-prefix'>
+                                {publicKey.replace('ed25519:', '').slice(0, 16)}
+                            </span>
+                            &nbsp;
+                            (<Translate id={`walletMigration.cleanKeys.keyTypes.${kind}`} />)
+                        </div>
+                        <div className='remove-checkbox' onClick={() => selectKey(publicKey)}>
+                            <Checkbox checked={selectedKeys[publicKey]} />
```

Comment:
recovery phrase type key should be unchecked and disabled:
![Uploading image.png…]()

Reply:
@hcho112 where is this requirement documented?

Reply:
@andy-haynes my apologies, it should have been updated on the ticket. (It was briefly mentioned in the comment section of the ticket as `- Newly created FAK from rotating keys step are now stored only in memory. (on WalletMigration.jsx, variable called rotatedKeys) So you may need to reference it on clean keys step to NOT delete this one. (Perhaps hide them from UI?)` but even this isn't clear enough.

I just came up with this requirement as that if we created a new FAK from rotating and aim to use this to pass down to other wallets, it make sense for us to make sure this is not deleted.

Reply:
Ah the `Recovery Phrase` is not the rotated key - it's the key the wallet has designated as its seed phrase. It's there if the user created the account with a seed phrase via wallet UI or if they added a seed phrase later on. We actually only know it's the `Recovery Phrase` because it's listed in DynamoDB as a recover method. A new FAK added from a BIP39 phrase via CLI or anywhere else outside the wallet UI would not qualify.

The rotated key I've opted not to show at all. I agreed with your comment around hiding it from the UI and implemented it accordingly, as its presence feels a bit confusing; i.e. I'm considering at the subset of _removable_ keys, not the set of keys on my account. I'm good either way though, just the decision I made in the absence of a definitive approach.